mxcube3/__init__.py

```diff
@@ -4,32 +4,42 @@
 import os, sys
 
 socketio = SocketIO()
-app = Flask(__name__, static_url_path='')
+app = Flask(__name__, static_url_path='')	
 app.debug = True
+socketio.init_app(app) # this line important for socketio msg, otherwise no msg is sent...
+@socketio.on('connect', namespace='/test')
+def connect():
+    print 'someone connected'
+    socketio.emit('test', {'data': 'Welcome'}, namespace='/test')
 
 opt_parser = OptionParser()
 opt_parser.add_option("-r", "--repository", 
                       dest="hwr_directory", 
                       help="Hardware Repository XML files path", 
                       default=os.path.join(os.path.dirname(__file__), 'HardwareObjects.xml/'))
+opt_parser.add_option("-l", "--log-file", 
+                      dest="log_file", 
+                      help="Hardware Repository log file name", 
+                      default=os.path.join(os.path.dirname(__file__), 'log/mxcube3.log'))
 cmdline_options, args = opt_parser.parse_args()
 
 ###Initialization of the HardwareObjects
-from .HardwareRepository import HardwareRepository 
+from .HardwareRepository import HardwareRepository, setLogFile
 HardwareRepository.addHardwareObjectsDirs([os.path.join(os.path.dirname(__file__), 'HardwareObjects')])
 # this is to allow Hardware Objects to do 'from HardwareRepository import ...'
-sys.path.insert(0, os.path.dirname(__file__)) 
+sys.path.insert(0, os.path.dirname(__file__))
 
 hwr_directory = cmdline_options.hwr_directory
+hwr_directory = './mxcube3/HardwareObjects.xml/'
```

Comment:
Sorry to bother again... In my case I use existing xml files from another directory ; would it be ok if you start mxcube3-server with `-r` option instead of hardcoding the directory ?